### PR TITLE
Add Active Python Releases to Downloads Page

### PR DESCRIPTION
**Pull request by crwilcox**
#1302 raised that it would be nice to have information, similar to other languages, that clearly states which python versions are actively maintained.

image

### CODE REVIEW

**Review by Mariatta — APPROVED** [review]
The table looks good to me. Thank you! I think this will help the community greatly.

**Comment by Mariatta** [comment]
@berkerpeksag will you be able to review and also deploy this to prod? Or if you can tell me how I can deploy to production, then I'll be able to help.

**Comment by ewdurbin** [comment]
Thank you @crwilcox for the implementation and @Mariatta for the idea and review!

My only concern here is with populating the table staticly in the template will require someone to update the template as new releases of python come into existence.

The @python/python-release-managers would probably be best suited to maintain this information as they already interact with the [`Release`](https://github.com/python/pythondotorg/blob/6902c19434763603f405061c31fb660454b1dbcd/downloads/models.py#L39) objects when performing their duties.

I'm not positive if it's worth blocking this on these concerns, but adding a new `Version` model to replace the very rough [`versions`](https://github.com/python/pythondotorg/blob/6902c19434763603f405061c31fb660454b1dbcd/downloads/models.py#L44-L52) field on `Release` would make this far more maintainable as part of a RM's role, and allow the list to be populated from the database rather than being hard coded in the template.

I'll leave some time for @python/python-release-managers to chime in with any thoughts, but otherwise this can be merged and the above ramblings converted into an issue :)

**Comment by crwilcox** [comment]
Thanks for taking a look @ewdurbin! I raised a similar concern on the issue. I think linking this to releases would be great. I wasn't quite sure the approach that would be most desired by release managers so I sort of stopped myself 😅

The coupling today of releases and updating PEPs/Dev Guide isn't super automated (as best I know anyway), but it all seems to work out. That said, I would be happy to take on getting a model setup for this, though as you said it might be easier to do that as a separate PR.

File: templates/downloads/index.html
Comment (berkerpeksag) [inline templates/downloads/index.html:41]: Is it possible to move this inside `<div class="row download-list-widget">`? If not, we need to remove the duplicate `render_active_banner` call in line 95.

File: templates/downloads/index.html
Comment (berkerpeksag) [inline templates/downloads/index.html]: Nitpick:

```suggestion
                    <p class="success-quote"><a href="https://devguide.python.org/#status-of-python-branches">For more information please visit the Python Developer's Guide</a>.</p>
```

File: templates/downloads/index.html
Comment (berkerpeksag) [inline templates/downloads/index.html]: Nitpick: Please delete the superfluous newlines.

```suggestion
```

**Review by berkerpeksag — COMMENTED** [review]
Thank you for the PR! In general, I agree with @ewdurbin that it would be better to find a more dynamic solution, but I think this is a good start for linking release schedule PEPs on www.python.org. If RMs will be happy about getting this in, it would be nice to update PEP 101 and include it as part of release process.

I left some comments about HTML and duplicate template tag.

File: templates/downloads/index.html
Comment (berkerpeksag) [reply templates/downloads/index.html:41]: Actually, you can ignore my comment about moving it inside `<div class="row download-list-widget">`. It looks pretty great! :) We just need to remove the duplicate `render_active_banner` call in line 95.

**Comment by ned-deily** [comment]
Thanks for the PR.  On behalf of the release managers, who maintain this information, I'd like to take a closer review of this over the next few days.

**Comment by crwilcox** [comment]
Thanks for the feedback @berkerpeksag. I think I have addressed all of it. @ned-deily have you had a chance to think about this? Are there other release folks worthing having take a look?

File: templates/downloads/index.html
Comment (hugovk) [inline templates/downloads/index.html]: Use sentence case for consistency with other table style:

```suggestion
                        <span class="release-version">Python version</span>
                        <span class="release-status">Maintenance status</span>
                        <span class="release-start">First released</span>
                        <span class="release-end">End of support</span>
                        <span class="release-pep">Release schedule</span>
```

File: templates/downloads/index.html
Comment (hugovk) [inline templates/downloads/index.html]: Remove "please" for brevity, cf. https://developers.google.com/style/tone#politeness-and-use-of-please

```suggestion
                    <p class="success-quote"><a href="https://devguide.python.org/#status-of-python-branches">For more information visit the Python Developer's Guide</a>.</p>
```

**Review by ned-deily — APPROVED** [review]
Sorry about the review delay!  I think it's a big improvement to the web site.  Thanks for the PR.  Obviously it's not ideal to have the release data hardwired into the templates but it will do for now.  Longer term, I'd like to find a way to extract the release information we have from the database and make it available as, say, a json file on the website that could then be used by multiple consumers including this web page.  But that's not a good reason to hold this up.  I will add a note to PEP 101, the Python Release Process document to remind release managers to update this data on the website at appropriate points in our release cycle.  Thanks again, everyone!

**Review by berkerpeksag — APPROVED** [review]
LGTM!

**Comment by ewdurbin** [comment]
Thanks @crwilcox! This is now live.

**Comment by ned-deily** [comment]
You know how you look at something too long and you overlook the obvious??  I've opened #1577 to change 2.7 status from bugfix to end-of-life (and then remove in the future).

**Comment by berkerpeksag** [comment]
Oops, I completely missed it when I looked at it before merging :)

**Comment by ned-deily** [comment]
We all did!  I guess it's still a bit hard to believe that 2.7 is gone :)

**Comment by crwilcox** [comment]
Thanks to everyone for helping get this in.

**Pull request merged**
